src/components/ha-selector/ha-selector-select.ts

```diff
@@ -26,25 +31,60 @@ export class HaSelectSelector extends LitElement {
   @property({ type: Boolean }) public required = true;
 
   protected render() {
-    if (this.required && this.selector.select.options!.length < 6) {
+    const options = this.selector.select.options.map((option) =>
+      typeof option === "object" ? option : { value: option, label: option }
+    );
+
+    if (this.selector.select.multiple) {
+      const value =
+        !this.value || this.value === "" ? [] : (this.value as string[]);
+
+      return html`
+        <ha-chip-set>
+          ${value?.map(
+            (item, idx) =>
+              html`
+                <ha-chip hasTrailingIcon>
+                  ${options.find((option) => option.value === item)?.label ||
+                  item}
+                  <ha-svg-icon
+                    slot="trailing-icon"
+                    .path=${mdiClose}
+                    .idx=${idx}
+                    @click=${this._removeItem}
+                  ></ha-svg-icon>
+                </ha-chip>
+              `
+          )}
+        </ha-chip-set>
+        <ha-combo-box
```

Comment:
Should it be a textbox if we allow custom values and there are no options?

Reply:
Actually, how does a user select a custom value in a combobox?

Reply:
They click away or enter

Reply:
Clicking away to select something is unexpected. Enter in a form generally submits the whole form so users don't expect it is handled by the current focused component.

Selecting an option from the dropdown works as a way to indicate you pick something.

Reply:
https://i.gyazo.com/f718e49b23c294d80548f0075a9aa448.mp4

I was mimicking the email field

Reply:
So the difference with the email field is that the text is converted to a chip in-place. That makes it look like you're creating a chip